src/puppetlabs/dujour/version_check.clj

```diff
@@ -88,10 +88,15 @@
                          (merge version-data)
                          json/generate-string)
         url update-server-url
-        {:keys [status body] :as resp} (client/post url
-                                         {:headers {"Accept" "application/json"}
-                                          :body request-body
-                                          :as :text})]
+        {:keys [status body] :as resp} (try
+                                         (client/post url
+                                                     {:headers {"Accept" "application/json"}
+                                                      :body request-body
+                                                      :as :text})
+                                         (catch java.io.IOException e
+                                           (throw+ {:type ::connection-error
+                                                    :message (.getMessage e)
+                                                    :exception e})))]
```

Comment:
Do we want to follow the PE style guide for exceptions? Usually it contains a namespaced `kind` error, a `msg` and an optional `details` key.